Client/Frontend/Widgets/PhotonActionSheetProtocol.swift

```diff
@@ -159,6 +161,28 @@ extension PhotonActionSheetProtocol {
             }
         }
         
+        let pinToTopSites = PhotonActionSheetItem(title: Strings.PinTopsiteActionTitle, iconString: "action_pin") { action in
+            guard let url = tab.url?.displayURL else { return }
+            let absoluteString = url.absoluteString
+            
+            let sql = self.profile.favicons as! SQLiteHistory
+            sql.getSitesForURLs([absoluteString]).uponQueue(.main) { result in
+                guard let results = result.successValue else {
+                    log.warning("Could not get site for \(absoluteString)")
+                    return
+                }
+                
+                if case let site?? = results[0] {
+                    self.profile.history.addPinnedTopSite(site).uponQueue(.main) { addResult in
```

Comment:
Normally we use uponQueue if we later need to do some sort of UI work. In this case I think you an avoid using `.main` here and above